api/worker_manager.go

```diff
@@ -0,0 +1,238 @@
+package api
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/vocdoni/davinci-node/log"
+)
+
+// BanRules defines the rules for banning workers. It includes the duration
+// for which a worker is banned and the maximum number of consecutive failed
+// jobs before a worker is banned.
+type BanRules struct {
+	BanTimeout          time.Duration // Duration for which the worker is banned
+	FailuresToGetBanned int           // Maximum consecutive failed jobs before banning
+}
+
+// DefaultBanRules provides the default ban rules for workers
+var DefaultBanRules = &BanRules{
+	BanTimeout:          3 * time.Minute, // Ban for 3 minutes
+	FailuresToGetBanned: 5,               // 3 consecutive failed jobs
+}
+
+// worker represents a worker that processes jobs
+type worker struct {
```

Comment:
I think the worker and worker manager, jobs manager, etc. must be moved to a separate package. It is not part of the "api" itself but used by the api package, and it has become too big.